From 1fb995afee86d320ca3a5ca7d802ff8807d83d38 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 15 Dec 2022 16:53:36 +0100
Subject: [PATCH 363/529] ARM: dts: add rsc table memory region on
 stm32mp157x-ed1 boards

Add the mcu_rsc_table region to support the resource table and the trace
buffer allocation in the MCU SRAM3 for signed firmware. by default the
MCU SRAM3 is shared between the Cortex-A and the cortex-M.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I0bac87f365aafa51e10a2144dbfa033dc1a74fe3
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/283832
---
 arch/arm/boot/dts/stm32mp157a-ed1.dts  | 8 +++++++-
 arch/arm/boot/dts/stm32mp157c-ed1.dts  | 8 +++++++-
 arch/arm/boot/dts/stm32mp157d-ed1.dts  | 8 +++++++-
 arch/arm/boot/dts/stm32mp157f-ed1.dts  | 8 +++++++-
 arch/arm/boot/dts/stm32mp15xx-dkx.dtsi | 2 +-
 5 files changed, 29 insertions(+), 5 deletions(-)

--- a/arch/arm/boot/dts/stm32mp157a-ed1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-ed1.dts
@@ -61,6 +61,12 @@
 			no-map;
 		};
 
+		mcu_rsc_table: mcu-rsc-table@10048000 {
+			compatible = "shared-dma-pool";
+			reg = <0x10048000 0x8000>;
+			no-map;
+		};
+
 		mcuram: mcuram@30000000 {
 			compatible = "shared-dma-pool";
 			reg = <0x30000000 0x40000>;
@@ -331,7 +337,7 @@
 
 &m4_rproc {
 	memory-region = <&retram>, <&mcuram>, <&mcuram2>, <&vdev0vring0>,
-			<&vdev0vring1>, <&vdev0buffer>;
+			<&vdev0vring1>, <&vdev0buffer>, <&mcu_rsc_table>;
 	mboxes = <&ipcc 0>, <&ipcc 1>, <&ipcc 2>, <&ipcc 3>;
 	mbox-names = "vq0", "vq1", "shutdown", "detach";
 	interrupt-parent = <&exti>;
--- a/arch/arm/boot/dts/stm32mp157c-ed1.dts
+++ b/arch/arm/boot/dts/stm32mp157c-ed1.dts
@@ -57,6 +57,12 @@
 			no-map;
 		};
 
+		mcu_rsc_table: mcu-rsc-table@10048000 {
+			compatible = "shared-dma-pool";
+			reg = <0x10048000 0x8000>;
+			no-map;
+		};
+
 		mcuram: mcuram@30000000 {
 			compatible = "shared-dma-pool";
 			reg = <0x30000000 0x40000>;
@@ -343,7 +349,7 @@
 
 &m4_rproc {
 	memory-region = <&retram>, <&mcuram>, <&mcuram2>, <&vdev0vring0>,
-			<&vdev0vring1>, <&vdev0buffer>;
+			<&vdev0vring1>, <&vdev0buffer>, <&mcu_rsc_table>;
 	mboxes = <&ipcc 0>, <&ipcc 1>, <&ipcc 2>, <&ipcc 3>;
 	mbox-names = "vq0", "vq1", "shutdown", "detach";
 	interrupt-parent = <&exti>;
--- a/arch/arm/boot/dts/stm32mp157d-ed1.dts
+++ b/arch/arm/boot/dts/stm32mp157d-ed1.dts
@@ -61,6 +61,12 @@
 			no-map;
 		};
 
+		mcu_rsc_table: mcu-rsc-table@10048000 {
+			compatible = "shared-dma-pool";
+			reg = <0x10048000 0x8000>;
+			no-map;
+		};
+
 		mcuram: mcuram@30000000 {
 			compatible = "shared-dma-pool";
 			reg = <0x30000000 0x40000>;
@@ -339,7 +345,7 @@
 
 &m4_rproc {
 	memory-region = <&retram>, <&mcuram>, <&mcuram2>, <&vdev0vring0>,
-			<&vdev0vring1>, <&vdev0buffer>;
+			<&vdev0vring1>, <&vdev0buffer>, <&mcu_rsc_table>;
 	mboxes = <&ipcc 0>, <&ipcc 1>, <&ipcc 2>, <&ipcc 3>;
 	mbox-names = "vq0", "vq1", "shutdown", "detach";
 	interrupt-parent = <&exti>;
--- a/arch/arm/boot/dts/stm32mp157f-ed1.dts
+++ b/arch/arm/boot/dts/stm32mp157f-ed1.dts
@@ -61,6 +61,12 @@
 			no-map;
 		};
 
+		mcu_rsc_table: mcu-rsc-table@10048000 {
+			compatible = "shared-dma-pool";
+			reg = <0x10048000 0x8000>;
+			no-map;
+		};
+
 		mcuram: mcuram@30000000 {
 			compatible = "shared-dma-pool";
 			reg = <0x30000000 0x40000>;
@@ -343,7 +349,7 @@
 
 &m4_rproc {
 	memory-region = <&retram>, <&mcuram>, <&mcuram2>, <&vdev0vring0>,
-			<&vdev0vring1>, <&vdev0buffer>;
+			<&vdev0vring1>, <&vdev0buffer>, <&mcu_rsc_table>;
 	mboxes = <&ipcc 0>, <&ipcc 1>, <&ipcc 2>, <&ipcc 3>;
 	mbox-names = "vq0", "vq1", "shutdown", "detach";
 	interrupt-parent = <&exti>;
--- a/arch/arm/boot/dts/stm32mp15xx-dkx.dtsi
+++ b/arch/arm/boot/dts/stm32mp15xx-dkx.dtsi
@@ -50,7 +50,7 @@
 			no-map;
 		};
 
-		mcu_rsc_table: mcu_rsc_table@10048000 {
+		mcu_rsc_table: mcu-rsc-table@10048000 {
 			compatible = "shared-dma-pool";
 			reg = <0x10048000 0x8000>;
 			no-map;
